src/ui/components/results/directanswercomponent.js

```diff
@@ -237,6 +219,112 @@ export default class DirectAnswerComponent extends Component {
     });
   }
 
+  /**
+   * Determines the card that should be used for the given direct answer.
+   *
+   * @param {Object} directAnswer The direct answer state
+   * @returns {string}
+   */
+  _getCard (directAnswer) {
+    if (this._types) {
+      return this._getCardBasedOnTypes(directAnswer);
+    } else if (this._cardOverrides.length > 0) {
+      return this._getCardBasedOnOverrides({
+        directAnswer: directAnswer,
+        overrides: this._cardOverrides,
+        fallback: this._defaultCard
+      });
+    } else {
+      return this._defaultCard;
+    }
+  }
+
+  /**
+   * Determines the card that should be used based on the types option
+   *
+   * @param {Object} directAnswer The direct answer state
+   * @returns {string}
+   */
+  _getCardBasedOnTypes (directAnswer) {
+    return Object.entries(this._types).reduce((cardType, [directAnswerType, typeOptions]) => {
+      if (directAnswerType !== directAnswer.type) {
+        return cardType;
+      }
+
+      const cardFallback = typeOptions.cardType || this._defaultCard;
+
+      if (typeOptions.cardOverrides) {
+        this._validateCardOverrides(typeOptions.cardOverrides);
+        return this._getCardBasedOnOverrides({
+          directAnswer: directAnswer,
+          overrides: typeOptions.cardOverrides,
+          fallback: cardFallback
+        });
+      }
+
+      return cardFallback;
+    }, this._defaultCard);
+  }
+
+  /**
+   * Returns the custom card type that should be used for the given direct answer.
+   *
+   * @param {Object} directAnswer The direct answer state
+   * @param {Object[]} overrides The overrides to search through
+   * @param {string} fallback The card to return if no match is found
+   * @returns {string}
+   */
+  _getCardBasedOnOverrides ({ directAnswer, overrides, fallback }) {
+    const cardOverride = overrides.find(override => {
+      return this._overrideMatchesAnswer(override, directAnswer);
+    });
+    return cardOverride ? cardOverride.cardType : fallback;
+  }
+
+  /**
+   * Check whether a given cardOverride matches a given directAnswer.
+   *
+   * @param {Object} override
+   * @param {Object} directAnswer
+   * @returns {boolean}
+   */
+  _overrideMatchesAnswer (override, directAnswer) {
+    if (!Object.keys(directAnswer).length) {
+      return true;
+    }
+    const directAnswerPropeties = {
+      type: directAnswer.type,
+      entityType: directAnswer.relatedItem.data.type,
+      fieldName: directAnswer.answer.fieldName,
+      fieldType: directAnswer.answer.fieldType
+    };
+    for (let [propertyToMatch, propertyValue] of Object.entries(override)) {
+      if (propertyToMatch === 'cardType') {
+        continue;
+      }
+      if (directAnswerPropeties[propertyToMatch] !== propertyValue) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Validates the cardOverrides option within the types option.
+   * Throws an error if the cardOverrides contains the 'type' option.
```

Comment:
I'm probably missing context but why do we throw an error for this?

Reply:
This was called out in the spec of something that isn't supported. The "types" object is already keyed by the type of the direct answer, so it doesn't really make sense if we let people put another type within the cardOverrides. I asked Rose what to do in this situation and she said to throw an error.